pkg/net/ip_port.go

```diff
@@ -16,21 +16,28 @@ type NetworkPeerID struct {
 	IPNetwork IPNetwork
 }
 
-// IsValid checks if the ip:port pair is valid.
-func (e NetworkPeerID) IsValid() bool {
+// IsAddressValid checks if the ip:port pair of the Address is valid.
+// This does not check the validity of IPNetwork.
+func (e NetworkPeerID) IsAddressValid() bool {
 	return e.Address.IsValid()
 }
 
 // String returns a string representation of this ip:port pair.
 func (e NetworkPeerID) String() string {
```

Comment:
It could be interesting to include the prefix length in the string representation of an IPNetwork, with a '/' suffix, for instance.

Reply:
> 	if addr == "" {
>		addr = e.IPNetwork.IP().String()
>		isIPv6 = e.IPNetwork.IP().Family() == IPv6
>	}

Since addr is the .IP() part of the IPNetwork, then wouldn't it be just the IP itself ?

Meanwhile, I realize that something like : `223.42.0.0/16:1234` would look a bit odd. It would still enable us to understand which case we have when reading the logs.

Reply:
Let's suppose that we have an IPNetwork and no Address: the IPNetwork has {ip=1.2.3.0, prefixLen=24}. Then addr will contain the IP representation, and the final result looks like `1.2.3.0:<port>`

This is all very nitpick, but I would suggest that we end-up with `1.2.3.0/24:<port>` instead.

Reply:
Ah, I see :) I will check how current version of String is used and will add that if possible.

Reply:
I got that covered in [bbc702b](https://github.com/stackrox/stackrox/pull/9718/commits/bbc702bd28e215aaa0c7bc073ac3dbca8c016c3e) (also added unit test coverage)